validator/src/commands/set_public_address/mod.rs

```diff
@@ -1,11 +1,39 @@
 use {
-    crate::{admin_rpc_service, cli::DefaultArgs},
+    crate::{admin_rpc_service, cli::DefaultArgs, commands::FromClapArgMatches},
     clap::{App, Arg, ArgGroup, ArgMatches, SubCommand},
     std::{net::SocketAddr, path::Path, process::exit},
 };
 
+const COMMAND: &str = "set-public-address";
+
+#[derive(Debug, PartialEq)]
+pub struct SetPublicAddressArgs {
+    pub tpu_addr: Option<SocketAddr>,
+    pub tpu_forwards_addr: Option<SocketAddr>,
+}
+
+impl FromClapArgMatches for SetPublicAddressArgs {
+    fn from_clap_arg_match(matches: &ArgMatches) -> Self {
+        let parse_arg_addr = |arg_name: &str, arg_long: &str| -> Option<SocketAddr> {
+            matches.value_of(arg_name).map(|host_port| {
+                solana_net_utils::parse_host_port(host_port).unwrap_or_else(|err| {
+                    eprintln!(
+                        "Failed to parse --{arg_long} address. It must be in the HOST:PORT \
+                     format. {err}"
+                    );
+                    exit(1);
```

Comment:
Hard to unit test something that exits; let's hold here until we figure out what direction we want to go with error handling in https://github.com/anza-xyz/agave/pull/4923